nixos/lib/testing-python.nix

```diff
@@ -207,14 +199,16 @@ rec {
     , ...
     } @ t:
     let
-      nodes = qemu_pkg:
-        let
-          testScript' =
-            # Call the test script with the computed nodes.
-            if lib.isFunction testScript
-            then testScript { nodes = nodes qemu_pkg; }
-            else testScript;
+      # Instantiates test script with nodes, if it's a function (contract)
+      mkTestScript = qemu_pkg:
+        # Call the test script with the computed nodes.
+        if lib.isFunction testScript
+        then testScript { nodes = nodes qemu_pkg false; }
```

Comment:
```suggestion
        # To avoid an infinite recursion (nodes -> testScript -> nodes), we force disable
        # the effect of config.virtualisation.useNixStoreImage, so `testScript -> nodes`
        # becomes `testScript -> nodes2` where `nodes2` does not reference the `testScript`.
        then testScript { nodes = nodes qemu_pkg false; }
```